r/R/filesystem.R

```diff
@@ -384,7 +387,24 @@ S3FileSystem <- R6Class("S3FileSystem",
     region = function() fs___S3FileSystem__region(self)
   )
 )
-S3FileSystem$create <- function(anonymous = FALSE, ...) {
+S3FileSystem$create <- function(
+  anonymous = FALSE,
+  log_level = c("fatal", "error", "warn", "info", "debug", "trace", "off"),
+  ...) {
+  # Init S3 now only if log_level is set
+  # TODO: This is clunky
```

Comment:
I thought this looked clunky but wasn't sure how best to guard this code. It seems like the way you're supposed to use `rlang::arg_match` is to make the first option in your set of options the default and just write code like,

```r
S3FileSystem$create <- function(log_level = c("fatal", "error", "warn", "info", "debug", "trace", "off")) {
  log_level <- rlang::arg_match(log_level)
  InitS3(log_level)

  # continue
}
```

However we don't always want to call InitS3(), hence the reason for guarding it like I did. Any suggestions welcome.

Reply:
No worries at all. A stand-alone `s3_initialize` was how I did my first version of this PR but the discussion lead us down putting the functionality inside `S3FileSystem$create`. `options()` was also suggested before and is pretty good but has one downside, described in https://github.com/apache/arrow/pull/35398#issuecomment-1533867236. Maybe the downside I describe there is acceptable and we just document it?